torchbenchmark/score/compute_score.py

```diff
@@ -9,10 +9,42 @@
 import os
 import yaml
 
+from collections import defaultdict
 from tabulate import tabulate
+from torchbenchmark import list_models
 
-SPEC_FILE_DEFAULT = "score.yml"
-TARGET_SCORE_DEFAULT = 1000
+def generate_spec():
+    """
+    Helper function which constructs the spec dictionary by iterating
+    over the existing models. This API is used for generating the
+    default spec hierarchy configuration.
+    return type:
+        Returns a dictionary of type collections.defaultdict. Defaultdict
+        handles `None` or missing values and hence, no explicit `None` check
+        is required.
+    Arguments: `None`
+    Note: Only those models available inside the required_models list are
+          used to construct the default spec hierarchy.
+    """
+    spec = {'hierarchy':{'model':defaultdict(dict)}}
+    # These are the models required to generate the default spec.
+    # Please update this list if any new model needs to be a part of
+    # the default spec configuration. Before you update, please consult
+    # with someone in the benchmark team.
+    required_models = ['pytorch_mobilenet_v3', 'yolov3', 'attention_is_all_you_need_pytorch', \
```

Comment:
i realize now, that this way of listing just the models is less 'secure' than the previous way of listing each fully-qualified test name.  

for instance, if on monday 'yolov3' runs in cpu and gpu mode, but on tuesday someone disables it on cpu for some reason, we'll compute a subtly different weighting of the score without realizing it.

What if we take the 'required_tests' directly from the normalization file?

In the case of a 'plot sweep' the normalization file could be anything, but I think it's OK/expected that it is flexible since developers on master can be using the plot on different sets of models.

For CI, where we want to be using a locked down version, it's critical anyway that the 'golden file' be used for reference, and since that file has to be frozen in the repo anyway, we can just trust it as the source of truth for which models to expect.  What do you think?

Reply:
Please find my comments here:

> i realize now, that this way of listing just the models is less 'secure' than the previous way of listing each fully-qualified test name.
> 
> for instance, if on monday 'yolov3' runs in cpu and gpu mode, but on tuesday someone disables it on cpu for some reason, we'll compute a subtly different weighting of the score without realizing it.

> This PR will only replace what score.yml did. So `generate_spec` will only generate the default spec for only those models that were enabled as part of `score.yml`. So incase the user doesn't specify any `spec.yml` , this function will generate the default spec used for calculating the score. May be the name `generate_spec` is misleading. Renaming it to `generate_default_spec` is more appropriate. This will not alter any score calculation yet. 

> What if we take the 'required_tests' directly from the normalization file?
>  Does normalization file here refer to the user provided score.yml? If that's the case, then the required tests from the normalization file. 

> In the case of a 'plot sweep' the normalization file could be anything, but I think it's OK/expected that it is flexible since developers on master can be using the plot on different sets of models.
> 
> For CI, where we want to be using a locked down version, it's critical anyway that the 'golden file' be used for reference, and since that file has to be frozen in the repo anyway, we can just trust it as the source of truth for which models to expect. What do you think?
> This default spec will be the 'golden file' - the default locked down version. 

Does this make sense?